woocommerce/woocommerce-gateway-stripe
# feat: confirmation modal on test mode toggle

# Changes proposed in this Pull Request:

Helps with https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1835

Adds a confirmation modal when enabling/disabling test mode.
When enabling live mode, we need to check if the "live" keys are present. When they aren't we need to prompt the user to enter them.
Same is valid when switching from "live" mode to "test" mode, we need to check for the presence of the "test" keys.

This just sets up a modal, the rest of the work is done in https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1835

When live keys are not present:
![2021-10-04 11 48 22](https://user-images.githubusercontent.com/273592/135894375-98bde366-b24a-4769-959a-5a4f1c76d9e1.gif)

When both keys are present.
![2021-10-04 11 48 59](https://user-images.githubusercontent.com/273592/135894350-47eb8209-a33c-40dd-b369-fe68b89cf334.gif)

This also ensures that when the test mode/live mode flag change, we also re-fetch the account data (which might have changed in the meantime). We do this by invalidating the `getAccountData` cache:
![2021-10-04 14 32 09](https://user-images.githubusercontent.com/273592/135913068-ecf8dc4a-902a-4946-95fc-bbcb94a6a58b.gif)


# Testing instructions
- Ensure you have the `_wcstripe_feature_upe_settings` flag enabled
- Go to http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings
- When attempting to enable live mode, if the live mode keys are not present a modal is displayed
- When attempting to enable test mode, if the test mode keys are not present a modal is displayed
- If both live and test mode keys are present, a modal is not displayed
